crates/sui-json-rpc/src/read_api.rs

```diff
@@ -152,17 +152,27 @@ impl ReadApiServer for ReadApi {
             .get_executed_transaction_and_effects(digest)
             .await
             .tap_err(|err| debug!(tx_digest=?digest, "Failed to get transaction: {:?}", err))?;
-        let checkpoint = self
+        let checkpoint_seq_opt = self
             .state
             .database
             .get_transaction_checkpoint(&digest)
-            .map_err(|e| anyhow!("{e}"))?;
+            .map_err(|e| anyhow!("{e}"))?
+            .map(|(_epoch, seq)| seq);
+        let mut checkpoint = None;
+        if let Some(checkpoint_seq) = checkpoint_seq_opt {
+            checkpoint = Some(
+                self.get_checkpoint(CheckpointId::SequenceNumber(checkpoint_seq))
+                    .await?,
+            );
+        }
+        let checkpoint_timestamp = checkpoint.map(|c| c.timestamp_ms);
```

Comment:
Can all be done in a one sequence of maps. Specifically, you would want to not error out even if there is an error when reading checkpoint (just make the timestamp None), which I think is preferred.

Reply:
thanks, made a change while I firstly tried map with an async closure like below and saw an `async closures are unstable` error, any wisdom?
```
        let checkpoint = checkpoint_seq_opt
            .map(async move |seq| {
                let tmp = self.get_checkpoint(CheckpointId::SequenceNumber(seq))
                    .await?;
                tmp
            });
```